util/src/main/scala/com/scalableminds/util/io/ZipIO.scala

```diff
@@ -93,17 +93,20 @@ object ZipIO extends LazyLogging {
   def zip(sources: List[NamedStream], out: OutputStream)(implicit ec: ExecutionContext): Future[Unit] =
     zip(sources.toIterator, out)
 
-  def zip(sources: Iterator[NamedStream], out: OutputStream)(implicit ec: ExecutionContext): Future[Unit] =
+  def zip(sources: Iterator[NamedStream], out: OutputStream)(implicit ec: ExecutionContext): Future[Unit] = {
+    val zip = startZip(out)
     if (sources.nonEmpty) {
-      val zip = startZip(out)
       for {
         _ <- zipIterator(sources, zip)
         _ = zip.close()
+        _ = out.close()
```

Comment:
I _tend_ to think that zip.close also does out.close but I’d rather be safe than sorry here